### PR TITLE
Added the unverified role thing + a utility command

### PR DESCRIPTION
**Pull request by vsmart-06**
Please do not merge this pull request till the roles are created. A test also has to be done for this in the testing server as my MongoDB is not working.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -360,6 +372,11 @@ async def hasRole(member: discord.Member, role_name):
             return True
     return False
 
+async def getRoleId(role_name: str):
+    guild = bot.get_guild(GUILD_ID)
+    role = discord.utils.get(guild.roles, name = role_name)
```
Comment (Infiniticity) [inline app.py:377]:
```suggestion
    role = discord.utils.get(guild.roles, name=role_name)
```

**Review by Infiniticity — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (vsmart-06) [reply app.py:377]: bro who cares abt that it doesn't matter lol

Comment (Infiniticity) [reply app.py:377]: It is just for consistency.

File: app.py [hunk 2]
```diff
@@ -73,6 +73,16 @@ async def on_raw_reaction_add(reaction):
...
+        if await hasRole(user, "Stage 1 - Unverified"):
+            unverified_stage1 = guild.get_role(await getRoleId("Stage 1 - Unverified"))
+            await user.remove_roles(unverified_stage1)
+            unverified_stage2 = guild.get_role(await getRoleId("Stage 2 - Unverified"))
+            await user.add_roles(unverified_stage2)
+        elif await hasRole(user, "Stage 2 - Unverified"):
+            unverified_stage2 = guild.get_role(await getRoleId("Stage 2 - Unverified"))
+            await user.remove_roles(unverified_stage2)
+            verified = guild.get_role(await getRoleId("Verified"))
+            await user.add_roles(verified)
```
Comment (Infiniticity) [inline app.py:85]:
```suggestion
        if await hasRole(user, "Stage 1 - Unverified"):
            unverified_stage1 = await getRole("Stage 1 - Unverified")
            await user.remove_roles(unverified_stage1)
            unverified_stage2 = await getRole("Stage 2 - Unverified")
            await user.add_roles(unverified_stage2)
        elif await hasRole(user, "Stage 2 - Unverified"):
            unverified_stage2 = await getRole("Stage 2 - Unverified")
            await user.remove_roles(unverified_stage2)
            verified = await getRoleId("Verified")
            await user.add_roles(verified)
```

File: app.py [hunk 3]
```diff
@@ -235,14 +245,16 @@ async def on_guild_join(guild):
...
         channel = await member.create_dm()
         await channel.send(embed=embed1)
         welcome = bot.get_channel(930088940654956575)
         await welcome.send(f"Welcome {member.mention}! Please pick up your roles at <#1010112017178312755> and <#1009302501566205952> to access the server.")
+        unverified_stage1 = member.guild.get_role(await getRoleId("Stage 1 - Unverified"))
```
Comment (Infiniticity) [inline app.py:256]:
```suggestion
        unverified_stage1 = await getRole("Stage 1 - Unverified")
```

File: app.py [hunk 4]
```diff
@@ -360,6 +372,11 @@ async def hasRole(member: discord.Member, role_name):
             return True
     return False
 
+async def getRoleId(role_name: str):
+    guild = bot.get_guild(GUILD_ID)
+    role = discord.utils.get(guild.roles, name = role_name)
+    return role.id
```
Comment (Infiniticity) [inline app.py:378]:
```suggestion
async def getRole(role_name: str):
    guild = bot.get_guild(GUILD_ID)
    role = discord.utils.get(guild.roles, name=role_name)
    return role
```

**Review by Infiniticity — CHANGES_REQUESTED** [review]
You can reduce the number of API calls.

File: app.py (hunk 2 above)
Comment (vsmart-06) [reply app.py:85]: I dont think `getRole()` is a thing

File: app.py (hunk 4 above)
Comment (vsmart-06) [reply app.py:378]: nvm missed this part

File: app.py (hunk 2 above)
Comment (Infiniticity) [reply app.py:85]: I have edited the function's name.

**Review by Infiniticity — APPROVED** [review]

**Review by Infiniticity — COMMENTED** [review]
You can remove this redundant line:
```python
view.message_id = message_id
```

File: app.py [hunk 5]
```diff
@@ -712,8 +730,7 @@ async def helper(
             embed = discord.Embed()
         embed.set_author(name=f"{str(interaction.user)}", icon_url=interaction.user.display_avatar.url)
         await interaction.send(helper_role.mention, embed=embed)
-        return
```
Comment (Infiniticity) [inline app.py:715]: Was this a mistake?

Comment (vsmart-06) [reply app.py:715]: Idk bro it was already there I did not add it

Comment (Infiniticity) [reply app.py:715]: You have remove the `return`.

Comment (vsmart-06) [reply app.py:715]: Oh maybe that was an accident. I don't remember doing that lol sorry.

**Review by Infiniticity — APPROVED** [review]

**Review by Sachin-dot-py — COMMENTED** [review]
This implementation would just check that _any_ two roles are picked up, so even picking up server roles or two session roles would verify the user, against the intended use case that is at least one session role & one subject role. 

Plus, those users who take the NOT IGCSE role are not required to take subject roles so they would not be verified using this system. 

However, in the majority of cases, this system would work, hence I am approving these changes now. In the future, we could look at another implementation

**Pull request merged**